pubsub.go

```diff
@@ -209,6 +217,11 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 
 	go ps.processLoop(ctx)
 
+	numcpu := runtime.NumCPU()
+	for i := 0; i < numcpu; i++ {
```

Comment:
can this `numcpu` be a configurable number?

Reply:
We could make it configurable, but is there any benefit in using more than `NumCPU` synchronous validation workers?

Reply:
If the validation function is not CPU-bound, yes. That’s the case for Eth2.0 folks.

Reply:
These validators will be asynchronous and be executed in a goroutine.
This is the number of workers for the validation front-end; it handles signature verification, synchronous validators (TBD), and spawning goroutines for async validators with the throttle.

Reply:
Added inline (synchronous) validation supprt.

Reply:
If we give a number less than `numcpu`, would that help other components of the program utilize  cpu resource better?

Reply:
hrm maybe. We can add an option to make it less of a hog.

Reply:
Added option to control the number of workers.